src/components/application_manager/doc/SDL.SDD.StateController.dox

```diff
@@ -0,0 +1,73 @@
+/**
+\page application_manager State Controller Detailed Design
+
+## State Controller Design
+- \subpage state_controller_hight_level_logic
+
+*/
+
+/**
+\page state_controller_hight_level_logic 1 State Controller hight level logic
+
+## Introduction
+The state controller is responsible for keeping applications inappropriate HMI states 
+HMI state contains: 
+ - HMILevel 
+ - AudioStreamingState
+ - VideoStreamingState
+ - SystemContext 
+ 
+SDL should take care of conflicts between the HMI state of different apps. 
+For example, 2 apps cannot be in FULL at the same time. So if one application goes FULL, another one should be moved to BACKGROUND or LIMITED.
+Rules are described in the scope of the proposal https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0150-video-streaming-state.md 
+
+SDL should keep resolve application HMI states to avoid conflicts. There are several triggers for HMI state change:
+ - application activation
+ - application deactivation
+ - Start System Event from HMI (Phone call, TTS, etc ...)
+ - End System Event from HMI 
+ - Resumption
+
+\link application_manager::StateController 
+\endlink is responcible for this.
+
+Event from HMI should change HMI state for app registered applications.
+Changing rules described in  https://github.com/smartdevicelink/sdl_evolution/blob/proposals/0150-video-streaming-state.md 
+Some modifications introduced within https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0264-Separating-the-change-of-Audible-status-and-the-change-of-HMI-Status.md 
+
+Events from HMI are temporary and will be ended.
```

Comment:
@LuxoftAKutsan probably its nice to mention here that temp events from HMI can be started by the following RPCs:
- VR.Started
- TTS.Started
- BC.OnEventChanged(is_active = true)

And can be stopped by the following RPCs:
- VR.Stopped
- TTS.Stopped
- BC.OnEventChanged(is_active = false)

Reply:
added